### PR TITLE
Fix tagged VlanInterface if attached to multiple vlan as untagged member

### PR DESCRIPTION
**Pull request by qiluo-msft**
#### Why I did it
Fix several bugs:
1. If one vlan member belongs to multiple vlans, and if any of the vlans is "Tagged" type, we respect the tagged type
2. If one vlan member belongs to multiple vlans, and all of the vlans have no "Tagged" type, we override it to be a tagged member
3. make sure `vlantype_name` is assigned correctly in each iteration

#### How to verify it
1. Test the command line to parse a minigraph and make sure the output does not change.
```
./sonic-cfggen -m minigraph.mlnx20.xml
```
The minigraph is for HwSKU Mellanox-SN2700-D40C8S8.

2. Test on a DUT with HwSKU Mellanox-SN2700-D40C8S8
```
sudo config load_minigraph
show vlan brief
```
Checked the "Port Tagging" column in the output.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106

### CODE REVIEW

**Review by ganglyu — DISMISSED** [review]

**Review by anish-n — DISMISSED** [review]

**Review by kktheballer — DISMISSED** [review]
Tested AzD scenario using this mg on latest public master image

**Comment by qiluo-msft** [comment]
Waiting @kktheballer to update unit test, and will combine into this PR.

**Review by kktheballer — DISMISSED** [review]
Validated 202012 version for AzD scenario

File: src/sonic-config-engine/tests/test_cfggen.py [hunk 1]
```diff
@@ -704,6 +709,9 @@ def test_minigraph_bgp_voq_chassis_peer(self):
     def test_minigraph_sub_port_interfaces(self, check_stderr=True):
         self.verify_sub_intf(check_stderr=check_stderr)
 
+    def test_minigraph_sub_port_intf_resource_non_backend_tor(self, check_stderr=True):
+        self.verify_sub_intf_non_backend_tor(graph_file=self.sample_resource_graph, check_stderr=check_stderr)
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/test_cfggen.py:713]: Why is it named "sub intf"?

File: src/sonic-config-engine/tests/test_cfggen.py [hunk 2]
```diff
@@ -737,6 +745,24 @@ def verify_no_vlan_member(self):
...
+
+        # INTERFACE table does not exist
+        # argument = '-m "' + graph_file + '" -p "' + self.port_config + '" -v "INTERFACE"'
+        # output = self.run_script(argument)
+        # self.assertEqual(output.strip(), "")
+
+        # PORTCHANNEL_INTERFACE table does not exist
+        # argument = '-m "' + graph_file + '" -p "' + self.port_config + '" -v "PORTCHANNEL_INTERFACE"'
+        # output = self.run_script(argument)
+        # self.assertEqual(output.strip(), "")
```
Comment (neethajohn) [inline src/sonic-config-engine/tests/test_cfggen.py:759]: remove

File: src/sonic-config-engine/tests/test_cfggen.py (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-config-engine/tests/test_cfggen.py:713]: I use `test_minigraph_sub_port_intf_resource_type` implementation as a reference.

**Review by anish-n — APPROVED** [review]

**Review by neethajohn — APPROVED** [review]

**Comment by qiluo-msft** [comment]
@prsunny Do you want this fix on 201911?

**Pull request merged**